### PR TITLE
setup project using create-react-app

### LINKED ISSUE #1: Set up repo with framework & build system
Use [Create React App](https://create-react-app.dev/) with:
- Typescript enabled
- Sass enabled
- polyfills for older browsers via [react-app-polyfill](https://github.com/facebook/create-react-app/blob/master/packages/react-app-polyfill/README.md)

[According to the docs](https://create-react-app.dev/docs/getting-started), I think the above should be achievable by doing:

```bash
npx create-react-app my-app --template typescript 
yarn add node-sass react-app-polyfill
```

**Up for discussion:**

- One caveat will be that the above command will install everything to `my-app` and not this repo `het-frontend`. We may need to `cp -r` from the CRA output dir to this repo as I'm not sure if we can use an existing dir with the `npx create-react-app` cmd.
- We should pin a NodeJS version, probably whatever the latest LTS version is, via the `package.json` `engine` field or a `.nvmrc` file if using Node Version Manager.
- We should decide on whether or not to use `npm` or `yarn` as a package manager. CRA uses `yarn` by default but by passing the `--use-npm` flag when running the above cmd we can default to `npm`.
- Code formatting: are we okay with using [Prettier](https://prettier.io/) to automatically format our files? (this can be [set up to happen on a git commit hook](https://create-react-app.dev/docs/setting-up-your-editor/#formatting-code-automatically))
- add an [.editorconfig](https://editorconfig.org/) file to help normalize various IDEs & code editors? Prettier may take care of this as well?

### PR DESCRIPTION
**Pull request by clhenrick**
Fixes #1

### CODE REVIEW

File: app/README.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+This project was bootstrapped with [Create React App](https://github.com/facebook/create-react-app).
```
Comment (kkatzen) [inline app/README.md:1]: Do we want to keep this README? It could be confusing to have two

File: app/src/index.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import React from "react";
+import ReactDOM from "react-dom";
+import "./styles/index.scss";
+import App from "./App";
+import * as serviceWorker from "./serviceWorker";
```
Comment (kkatzen) [inline app/src/index.tsx:5]: Do we want to keep the service worker?

**Review by kkatzen — APPROVED** [review]
Thanks!

File: app/src/index.tsx (hunk 2 above)
Comment (aaronsn) [reply app/src/index.tsx:5]: So.... my job for the last ~4 years has been working on the Docs/Sheets/Slides service worker. I would really discourage using it unless we find a strong need for it (eg we need offline support or we encounter performance issues that can't be solved in a simpler way). It can be a big maintenance/debugging challenge, can cause performance issues, and has a pretty high ramp-up cost for devs to understand how it works.

TL;DR I think we should drop this

Comment (clhenrick) [reply app/src/index.tsx:5]: sounds good to me @aaronsn

File: app/README.md (hunk 1 above)
Comment (clhenrick) [reply app/README.md:1]: If we remove this readme then we might want to mention that the project was set up using CRA in the root readme

**Review by aaronsn — APPROVED** [review]

**Comment by aaronsn** [comment]
This is awesome, thanks for setting it up!

File: app/src/index.tsx (hunk 2 above)
Comment (clhenrick) [reply app/src/index.tsx:5]: sure, I will remove it.

File: app/public/index.html [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <meta name="theme-color" content="#000000" />
+    <meta
+      name="description"
+      content="Health Equity Tracker is a free-to-use data and visualization
+      platform that is enabling new insights into the impact of COVID-19 on
+      marginalized groups in the United States."
+    />
+    <link rel="apple-touch-icon" href="%PUBLIC_URL%/logo192.png" />
```
[the remark this answers is not in the document]
Comment (clhenrick) [reply app/public/index.html:14]: Yeah, they're boilerplate and are in the `app/public` dir. I propose leaving them for now just so we know what sizes they are at and how to swap them out later. We should also ask Ken to make a favicon.

[the remark this answers is not in the document]
Comment (aaronsn) [reply app/public/index.html:14]: SGTM

File: app/README.md (hunk 1 above)
Comment (clhenrick) [reply app/README.md]: Moved CRA docs to the root level readme and removed the CRA readme.

**Comment by clhenrick** [comment]
@kkatzen @aaronsn just FYI that I added a few more commits, they should be self explanatory but let me know if you have questions.

**Comment by aaronsn** [comment]
Looks good, you can go ahead and submit

File: README.md [hunk 4]
```diff
@@ -1 +1,67 @@
...
+
+To create a "production" build do:
+
+```bash
+npm run build
+```
+
+This should output bundled files in the `app/build/` directory. Use these files for hosting the web app.
+
+### Ejecting CRA
```
Comment (kkatzen) [inline README.md:55]: maybe spell out Create React App just to be extra clear?

File: app/src/App.test.tsx [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+import App from './App';
+
+test('renders learn react link', () => {
```
Comment (kkatzen) [inline app/src/App.test.tsx:5]: Maybe update to the new expected text "Health Equity Tracker Frontend"

**Review by kkatzen — APPROVED** [review]
I was able to run everything just fine with README :)

**Comment by aaronsn** [comment]
Chris - I noticed the lint-staged documentation suggests installing via `npm install --save-dev`. Should we be doing this for dev dependencies so they don't show up in the production build? Or does create-react-app handle that automatically?

**Comment by clhenrick** [comment]
> Chris - I noticed the lint-staged documentation suggests installing via `npm install --save-dev`. Should we be doing this for dev dependencies so they don't show up in the production build? Or does create-react-app handle that automatically?

@aaronsn I think it's handled automatically. I verified by doing the following as outlined in the CRA docs for [analyzing bundle size](https://create-react-app.dev/docs/analyzing-the-bundle-size)

```bash
# create a prod build
npm run build

# analyze the JS in the build dir
npx source-map-explorer build/static/js/*.js
``

This gave me the following chart / UI in the browser:

![image](https://user-images.githubusercontent.com/161748/95522546-36d08b00-0981-11eb-8978-4aeb80b12603.png)

As you can see the combined output is 128kb, so I don't see how other dependencies like node-sass, prettier, etc. could be included. Also in the docs when it mentions installing dev deps it doesn't say to specify them as such, though maybe this is just for dev deps that CRA anticipates? Probably worth investigating further to be on the safe side, but right now it seems to be okay.
